### PR TITLE
feat: titus starship prompt setup script

### PR DESCRIPTION
**Pull request by harshv5094**
## Type of Change
- [x] New feature

## Description
If anyone just want to add titus prompt style on their config. This script sets up the prompt without installing whole titus bash prompt.

## Testing
- Test are performed on my own system in bash and z shell.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/tab_data.toml [hunk 1]
```diff
@@ -74,7 +74,12 @@ name = "Rofi"
 description = "Rofi is a window switcher, run dialog, ssh-launcher and dmenu replacement that started as a clone of simpleswitcher, written by Sean Pringle and later expanded by Dave Davenport.\nThis command installs and configures rofi with configuration from CTT's DWM repo.\nhttps://github.com/ChrisTitusTech/dwm-titus"
 script = "rofi-setup.sh"
 
+[[data]]
+name = "Titus Starship Prompt Setup"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/tab_data.toml:78]:
```suggestion
name = "Titus Starship Prompt"
```

File: core/tabs/applications-setup/tab_data.toml [hunk 2]
```diff
@@ -47,7 +47,7 @@ script = "linutil-installer.sh"
 [[data.preconditions]]
 matches = false
 data = "command_exists"
-values = [ "linutil" ]
+values = ["linutil"]
```
Comment (adamperkowski) [inline core/tabs/applications-setup/tab_data.toml]: Why are you changing this? How is this relevant to this PR?

File: core/tabs/applications-setup/tab_data.toml [hunk 3]
```diff
@@ -57,7 +57,7 @@ script = "linutil-updater.sh"
 [[data.preconditions]]
 matches = true
 data = "command_exists"
-values = [ "linutil" ]
+values = ["linutil"]
```
Comment (adamperkowski) [inline core/tabs/applications-setup/tab_data.toml]: Why are you changing this? How is this relevant to this PR?

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+setupStarshipTomlFile() {
+  printf "%b\n" "${YELLOW} Copying starship prompt config ${RC}"
+  if [ -d "$HOME/.config/starship" ]; then
+    printf "%b\n" "${CYAN} Backing up existing starship prompt ${RC}"
+    mv "$HOME/.config/starship/" "$HOME/.config/starship.bk/"
+    cmdCheck
+    printf "%b\n" "${CYAN} Download Titus starship prompt file ${RC}"
+    mkdir -p "$HOME/.config/starship/"
+    curl -sSLo "${HOME}/.config/starship/starship.toml" "https://github.com/ChrisTitusTech/mybash/raw/main/starship.toml"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:33]:
```suggestion
    curl -sSLo "$HOME/.config/starship/starship.toml" "https://github.com/ChrisTitusTech/mybash/raw/main/starship.toml"
```

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    cmdCheck
+  else
+    printf "%b\n" "${CYAN} Download Titus starship prompt file ${RC}"
+    mkdir -p "$HOME/.config/starship/"
+    curl -sSLo "${HOME}/.config/starship/starship.toml" "https://github.com/ChrisTitusTech/mybash/raw/main/starship.toml"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:38]:
```suggestion
    curl -sSLo "$HOME/.config/starship/starship.toml" "https://github.com/ChrisTitusTech/mybash/raw/main/starship.toml"
```

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  fi
+}
+
+setupStarshipTomlFile() {
+  printf "%b\n" "${YELLOW} Copying starship prompt config ${RC}"
+  if [ -d "$HOME/.config/starship" ]; then
+    printf "%b\n" "${CYAN} Backing up existing starship prompt ${RC}"
+    mv "$HOME/.config/starship/" "$HOME/.config/starship.bk/"
+    cmdCheck
+    printf "%b\n" "${CYAN} Download Titus starship prompt file ${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:31]:
```suggestion
    printf "%b\n" "${CYAN} Downloading Titus' starship prompt file ${RC}"
```

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    mkdir -p "$HOME/.config/starship/"
+    curl -sSLo "${HOME}/.config/starship/starship.toml" "https://github.com/ChrisTitusTech/mybash/raw/main/starship.toml"
+    cmdCheck
+  else
+    printf "%b\n" "${CYAN} Download Titus starship prompt file ${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:36]:
```suggestion
    printf "%b\n" "${CYAN} Downloading Titus' starship prompt file ${RC}"
```

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  fi
+}
+
+# NOTE: Checking if starship config line exist in particular shell
+configureShell() {
+  printf "%b\n" "${CYAN} Setting up your shell. ${RC}"
+  if [ -e "$HOME/.bashrc" ]; then
+    printf "%b\n" "${CYAN} Setting up bash to use starship as default prompt ${RC}"
+    printf "%s\n" 'eval "$(starship init bash)"' >>"$HOME/.bashrc"
+    printf "%s\n" "export STARSHIP_CONFIG=~/.config/starship/starship.toml" >>"$HOME/.bashrc"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:49]:
```suggestion
    printf "%s\n" 'eval "$(starship init bash)"' >> "$HOME/.bashrc"
    printf "%s\n" "export STARSHIP_CONFIG=~/.config/starship/starship.toml" >> "$HOME/.bashrc"
```

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    cmdCheck
+  fi
+
+  if [ -e "$HOME/.zshrc" ]; then
+    printf "%b\n" "${CYAN} Setting up zsh to use starship as default prompt ${RC}"
+    printf "%s\n" 'eval "$(starship init zsh)"' >>"$HOME/.zshrc"
+    printf "%s\n" "export STARSHIP_CONFIG=~/.config/starship/starship.toml" >>"$HOME/.zshrc"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:56]:
```suggestion
    printf "%s\n" 'eval "$(starship init zsh)"' >> "$HOME/.zshrc"
    printf "%s\n" "export STARSHIP_CONFIG=~/.config/starship/starship.toml" >> "$HOME/.zshrc"
```

**Comment by jeevithakannan2** [comment]
Starship.toml will only work in ~/.config/starship.toml and not in ~/.config/staship/starship.toml

https://starship.rs/config/

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    cmdCheck
+  fi
+
+  # NOTE: There is a script in linutil application section which sets up zsh in .config directory.
+  # I need to add this config just for that script, if someone has configured their z shell using linutil
+  if [ -e "$HOME/.config/.zshrc" ]; then
+    printf "%b\n" "${CYAN} Setting up zsh to use starship as default prompt ${RC}"
+    printf "%s\n" 'eval "$(starship init fish)"' >>"$HOME/.config/.zshrc"
+    printf "%s\n" "export STARSHIP_CONFIG=~/.config/starship/starship.toml" >>"$HOME/.config/.zshrc"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:65]:
```suggestion
    printf "%s\n" 'eval "$(starship init fish)"' >> "$HOME/.config/.zshrc"
    printf "%s\n" "export STARSHIP_CONFIG=~/.config/starship/starship.toml" >> "$HOME/.config/.zshrc"
```

File: core/tabs/applications-setup/titus-starship-prompt-setup.sh [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    cmdCheck
+  fi
+}
+
+checkEnv
+printf "%b\n" "${YELLOW}NOTE: currently this script only support bash and zsh."
+installStarship
+setupStarshipTomlFile
+configureShell
```
Comment (adamperkowski) [inline core/tabs/applications-setup/titus-starship-prompt-setup.sh:74]: Move those printfs to the functions.

**Review by adamperkowski — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/tab_data.toml (hunk 2 above)
Comment (harshv5094) [reply core/tabs/applications-setup/tab_data.toml]: sorry I've already fixed this. That was done by my text editor

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]
Please close this, it's already included in the mybash script. NO need for another script doing the SAME exact thing.

**Comment by adamperkowski** [comment]
I don'r feel like this addition is neccessary. We need to be more careful with how many scripts we add and start thinking more about performance. @ChrisTitusTech?

**Comment by harshv5094** [comment]
Okay I'll close this

**Comment by harshv5094** [comment]
> Please close this, it's already included in the mybash script. NO need for another script doing the SAME exact thing.

I'm just only installing starship prompt. Not the whole mybash setup @nnyyxxxx.

**Comment by nnyyxxxx** [comment]
starship is already included in the mybash-setup script you're just duplicating that script entirely, no need for a separate script doing the same exact thing. @harshv5094

**Comment by nnyyxxxx** [comment]
image

**Comment by adamperkowski** [comment]
> I'm just only installing starship prompt. Not the whole mybash setup @nnyyxxxx.

@nnyyxxxx is already implementing this in his other PR.

**Comment by harshv5094** [comment]
okay I'll close it
